File: problem_solve/08-11/[PGS] 입국심사/ye0ngeun.py
def solution(n, times):
    # 주어진 시간 안에 모든 심사관이 n명 이상을 처리할 수 있는지 확인
    def is_possible(best_time):
        total = 0
        for t in times:
            # 심사관 한 명이 best_time 동안 처리할 수 있는 사람 수
            total += best_time // t
        return total >= n
    
    # 정답(최소 심사 시간)을 이분 탐색
    # 가장 느린 심사관이 혼자 n명을 처리하는 시간을 최대 범위로 설정
    lo, hi = 1, max(times) * n
    answer = hi
    
    while lo <= hi:
        mid = (lo + hi) // 2
        if is_possible(mid):
            # mid 시간 안에 처리가 가능하므로
            # 현재 값을 후보로 저장하고 더 짧은 시간이 가능한지 탐색
            answer = mid
            hi = mid -1
        else:
            # mid 시간으로는 부족하므로 더 긴 시간 탐색
            lo = mid + 1
    return answer

File: problem_solve/08-12/[PGS] 정수 삼각형/ye0ngeun.java
class Solution {
    public int solution(int[][] triangle) {
        for (int i = 1; i < triangle.length; i++) {
            for (int j = 0; j < triangle[i].length; j++) {
                // 맨 왼쪽
                if (j == 0) {
                    triangle[i][j] += triangle[i-1][j];
                } 
                // 맨 오른쪽
                else if (j == i) {
                    triangle[i][j] += triangle[i-1][j-1];
                }
                // 가운데
                else {
                    triangle[i][j] += Math.max(triangle[i-1][j-1], triangle[i-1][j]);
                }
            }
        }
        
        int[] lastRow = triangle[triangle.length - 1];
        int answer = 0;

        for (int value : lastRow) {
            answer = Math.max(answer, value);
        }
        
        return answer;
    }
}
